Fix #57 : Overhaul real-time communications page + add Session messenger

lrq3000 opened the pull request:
## Description

Resolves: #57, #138, #50

* Add audits systematically whenever available.
* Separate descriptions of networks types in a separate tutorial section
* Tag by network types instead of sections only for instant messengers)
* Add illustrations of networks types.
* Rework a bit the definitions and pros/cons.
* Add Anonymous Routing as a network type.
* Add Session (ex Loki Messenger)
* Upgrade Status messenger (ex status.im messenger) into a recommendation (needs more testing to confirm).

Following discussions below:
* Remove non audited or non E2EE apps: Jami, Mumble.
* Add lots of warning as notes for almost all software messengers.
* Migrate template from legacy to evergreen
* Add labels to evergreen template (more flexible alternative to info and warning items, necessary to implement the tags based classification of messengers by networks types)

Todo before merge:
- [ ] Check if Status is worthy of being a recommendation (usability, security concerns since after the 2019 audit, etc).
- [ ] Consider Jami as a videoconferencing recommendation (need to confirm how it E2EE encrypts group calls).

This is a proposition, to overhaul the real-time communications page.

Currently, instant messengers are separated in different categories. This forces to dichotomize messengers into one category or the other, which doesn't fit well for some messengers that implement multiple networks (eg, Briar which is both P2P and Anonymous Routing). Furthermore, only instant messengers are identified by their network type, but this should also apply to video and voice messengers as well as teamchat.

I propose here following the discussion in #57 to redesign the page by removing the network type sections, and instead just present a list of instant messengers, but with tags in each description instead of sections. This allows for a more streamlined overview, similar to other listings in other pages, while not losing any information. In fact, there are more infos now since we can assign multiple networks types.

The layman descriptions of the differences between each network type is not lost either, but moved at the bottom of the page. It could be moved at the top too depending on what you want to present first (the selection of apps or a tutorial to help the user choose). Illustrations have been added, as well as an Anonymous Routing network type.

Note: the PR may be buggy (I will check with Netlify, build toolchain fails on my computer for some reason...). I will update this post when fully ready.

/EDIT: somewhat ready now to discuss further. Probably needs some more polishing if this proposition is accepted.

/EDIT2: Note that [Jami has not yet been audited](https://dt.gl/messnger-provider-review-2021-jami/).

/EDIT3: for future reference, all icons that can be used for tags are listed here: https://www.angularjswiki.com/fontawesome/

dngray commented:
I hadn't noticed this comprehensive PR, will give it a check over shortly.

tommytran732 commented:
This looks great, but I would like to suggest that we remove jami due to it being unaudited. Status is audited but that was back in 2019 and had a lot of unfixed issues back then, so I'd remove that as well.

lrq3000 commented:
@tommytran732 I agree that Jami not being audited is a malus. On the plus side, it's one of the only E2EE and P2P messengers. But in practice in my experience it doesn't work well anyway.

About status.im, I never used this software, but from the doc it seems to fit all criteria. If the audit being 2 years old is an issue, the same criterion should be applied to other messengers, and I'm not sure a lot would pass a bar this high...

I would also suggest to remove Mumble. Although it is a very beloved software, it does not implement E2EE, which is a necessary criterion per the introduction of this page. Or at least it should be a "Worth Mentioning", but not a recommendation.

About the advantages/disadvantages, they are still there, but at the bottom of the page in the tutorial section on the various network types. Most messengers nowadays implement multiple network types for various features (eg, voice and video calls being P2P while messenging can be centralized/federated/anonymously routed), so the pros/cons are necessarily contextual (ie, no pros/cons per software but rather per network type, so that for hybrid messengers implementing multiple network types, the pros/cons for each of these network types apply depending on what features of the messenger are being used). I will add a short explanation to clarify in the tutorial section.

dngray commented:
> Adding session and the audit tag looks great, but I would like to suggest that we remove jami due to it being unaudited. Status is audited but that was back in 2019 and had a lot of unfixed issues back then, so I'd remove that as well.

In line with https://github.com/privacyguides/privacyguides.org/discussions/24 we want to establish a criteria for this page too.

I think it's worth mentioning that all submissions must be audited competently. Unofficially we'd already had that as a criteria, besides being open source. E2EE being another one that came in when we added Matrix and that was imminently enabling that.

> I would also suggest to remove Mumble. Although it is a very beloved software, it does not implement E2EE, which is a necessary criterion per the introduction of this page. Or at least it should be a "Worth Mentioning", but not a recommendation.

Agreed.

> About the advantages/disadvantages, they are still there, but at the bottom of the page in the tutorial section on the various network types. Most messengers nowadays implement multiple network types for various features (eg, voice and video calls being P2P while messenging can be centralized/federated/anonymously routed), so the pros/cons are necessarily contextual (ie, no pros/cons per software but rather per network type, so that for hybrid messengers implementing multiple network types, the pros/cons for each of these network types apply depending on what features of the messenger are being used). I will add a short explanation to clarify in the tutorial section.

This section may need re-working to what is currently there, as there is a bit of crossover now.

lrq3000 commented:
> This section may need re-working to what is currently there, as there is a bit of crossover now.

Yes the text needs polishing, but I'm not sure I understand what you have in mind. Could you please provide me with an example of what you would like to change?

dngray commented:
> > This section may need re-working to what is currently there, as there is a bit of crossover now.
> 
> Yes the text needs polishing, but I'm not sure I understand what you have in mind. Could you please provide me with an example of what you would like to change?

I had another look today and I'm not sure what I was talking about. :smile: 

We could get rid of the team section now, as Rocket Chat's E2EE hasn't been audited. With Matrix you can run a non-federating server, so that is suitable for companies too. Saves on duplication. You could mention that under the badge text for Federating with Element.

I think we can get rid of that WebRTC warning, that is no longer true. It isn't in Firefox nor Chromium.

lrq3000 commented:
About Rocket.chat's audit, I found the following infos:
* Their "management system" is independently audited (but not their technical system): https://docs.rocket.chat/guides/security/compliance-resources
* The only independent audit was done with automatic tools: https://www.isgroup.it/node.security/audit-Rocket.Chat.html
* 7 months ago, an independent researcher and member of a digital rights movement asked for the public release of infos about an independent audit, and the Rocket.chat team only pointed to contacting them in private: https://giters.com/RocketChat/Rocket.Chat/issues/21859

This looks quite shady. So I will proceed to remove Rocket.chat .

About removing the entire Team Chat section, I think it's a bit sad because it's a very different use case but it's useless to keep an entire section if there is only one software inside. This will become a label for now.

What about the file structure? Currently the files are in "legacy" folders, but since this would be an almost complete rewrite, maybe we should move the files to reflect that this is no longer legacy? If so, what should be the files structure?

lrq3000 commented:
There's a problem with requiring independent audits, all softwares in Voice and Video Messengers are not independently audited. Maybe because the encrypted multiparty VoIP feature is still quite new and experimental (implemented in 2019 as LIME in Linphone, implemented in the latter half of 2020 for Jitsi Meet).

Should we make an exception for these, or remove them, or retract the "independently audited" criterion?

lrq3000 commented:
I would also suggest to make status.im a recommendation. I never used the software so a practical test would be necessary first, but the more I dig into the technicals, the more it appears to fit all our criteria, including independent audits. So I'm not sure why it's only "Worth Mentioning" currently, maybe because it uses a blockchain network?

lrq3000 commented:
An excellent tutorial on anonymous routing, maybe we could add it in a "Further reading" section in anonymous routing? https://medium.com/unitychain/privacy-blockchain-and-onion-routing-d5609c611841

dngray commented:
@lrq3000

Sorry I haven't replied earlier, as we've been busy updating other parts of the site/doign research :smile: 

> About Rocket.chat's audit, I found the following infos:
> 
> * Their "management system" is independently audited (but not their technical system): https://docs.rocket.chat/guides/security/compliance-resources
> 
> * The only independent audit was done with automatic tools: https://www.isgroup.it/node.security/audit-Rocket.Chat.html
> 
> * 7 months ago, an independent researcher and member of a digital rights movement asked for the public release of infos about an independent audit, and the Rocket.chat team only pointed to contacting them in private: https://giters.com/RocketChat/Rocket.Chat/issues/21859
> 
> This looks quite shady. So I will proceed to remove Rocket.chat .

Indeed, these things should be public, and audits aren't using automatic tools.

> About removing the entire Team Chat section, I think it's a bit sad because it's a very different use case but it's useless to keep an entire section if there is only one software inside. This will become a label for now.

I don't think its worthwhile keeping for one piece of software.

What I think we could do is add a footnote for Element that Matrix servers when configured for business can have federation disabled if they are intended for internal company use only.

About that specifically, with the new format described below, longer descriptions are possible without sending the site layout into disarray (see hardened firefox for example).

> What about the file structure? Currently the files are in "legacy" folders, but since this would be an almost complete rewrite, maybe we should move the files to reflect that this is no longer legacy? If so, what should be the files structure?

About the file structure, basically the "legacy" is the PTIO style pages. The newer format will be individual YAML files for each card. You can see an example of where we're starting to do this on the browser section https://github.com/privacyguides/privacyguides.org/pull/309/files

What this means is then each "item" on the site actually has its data extracted from the main layout pages.

> There's a problem with requiring independent audits, all softwares in Voice and Video Messengers are not independently audited. Maybe because the encrypted multiparty VoIP feature is still quite new and experimental (implemented in 2019 as LIME in Linphone, implemented in the latter half of 2020 for Jitsi Meet).

> Should we make an exception for these, or remove them, or retract the "independently audited" criterion? (the latter implying Jami and Rocket.chat could be reintegrated?)

Personally I'm in favour of stripping them rather than making exceptions. The reason is it makes the site less complex, and encourages things to actually get an audit to be listed.

Otherwise we really open the floodgates to "everything"

> I would also suggest to make status.im a recommendation. I never used the software so a practical test would be necessary first, but the more I dig into the technicals, the more it appears to fit all our criteria, including independent audits. So I'm not sure why it's only "Worth Mentioning" currently, maybe because it uses a blockchain network?

A couple of reasons, I thik the first was a bug regarding notifications @JonahAragon tested this sime time ago.

The second was we were waiting for it to be listed in F-Droid which it now is: https://f-droid.org/en/packages/im.status.ethereum/

We're not against adding products with cryptocurrency features, we're just not open to adding *everything* that exists.

lrq3000 commented:
@dngray Thank you for your clarifications and suggestions. No worries, I know you guys are very busy since the fork, and anyway your replies were always timely imho (but I wouldn't mind if not, I'll try to follow through your schedule).

About being selective, I agree, and I try to consider softwares for PG only based on whether they fit all the criteria. Since the criteria are stringent, there are only a handful of softwares that can fit, so I think this takes objectively care of the selectivity problem.

I hence understand your suggestion to remove the voice/video calling softwares altogether, however this means that PG won't provide any alternative suggestion to Zoom and Microsoft Teams. IMHO using Jitsi Meet or Linphone is much better albeit not audited. Maybe we can make them a "Worth Mentioning" suggestion?

Indeed, we could consider the "Worth Mentioning" sections as a place to include softwares that do not (yet) meet all criteria, if that's ok for you. In this case, we will end up with a single section "Instant messengers", remove the Voice/Video Communication section, and add Jitsi Meet and Linphone as Worth Mentioning with a description that explains their purpose.

About status.im, then I will test it to see if the issues were resolved and if yes I will add it since it seems to fit all criteria.

dngray commented:
> @dngray Thank you for your clarifications and suggestions. No worries, I know you guys are very busy since the fork, and anyway your replies were always timely imho (but I wouldn't mind if not, I'll try to follow through your schedule).

That's okay, I do very much appreciate the help with PRs, especially regarding this page.

> About being selective, I agree, and I try to consider softwares for PG only based on whether they fit all the criteria. Since the criteria are stringent, there are only a handful of softwares that can fit, so I think this takes objectively care of the selectivity problem.

It does, and I think honestly more is not better as it clutters the page and the user then asks "which one do I pick?"

> I hence understand your suggestion to remove the voice/video calling softwares altogether, however this means that PG won't provide any alternative suggestion to Zoom and Microsoft Teams. IMHO using Jitsi Meet or Linphone is much better albeit not audited. Maybe we can make them a "Worth Mentioning" suggestion?

I was only suggesting removing the team category as it only had Element in it.

> Indeed, we could consider the "Worth Mentioning" sections as a place to include softwares that do not (yet) meet all criteria, if that's ok for you.

We're moving away from "Worth Mentioning" sections, we can always revist those softwares in a [discussion thread](https://github.com/privacyguides/privacyguides.org/discussions).

> In this case, we will end up with a single section "Instant messengers", remove the Voice/Video Communication section, and add Jitsi Meet and Linphone as Worth Mentioning with a description that explains their purpose.

Maybe a "Video Conferencing" section would work?

> About status.im, then I will test it to see if the issues were resolved and if yes I will add it since it seems to fit all criteria.

Would it be too much to ask if you take a look at the new format before we merge this.

An example can be seen in https://github.com/privacyguides/privacyguides.org/pull/309 and https://github.com/privacyguides/privacyguides.org/pull/333

We might as well look at it now, before merging in legacy.

lrq3000 commented:
@dngray I had a look at the new evergreen template thanks to the PR pointers you gave me. Wow, that's a great departure from the previous template! But it's great to see PG heading towards a more tutorial oriented approach, rather than listing. I understand the idea.

I updated this PR to migrate to the evergreen template, please let me know what you think about it. I had to also migrate the labels subtemplate to allow the implementation of tags.

I upgraded Status to a recommendation temporarily because I just moved everything to the evergreen template (and there is no Worth Mentions in it), I still need to test it out for myself.

About removing the Videoconferencing section, I was referring to the fact that if independent audits are required for all recommended softwares, none of those in the videoconferencing section would pass, hence the section would be empty and hence would need to be removed. So what do we do about that?

tommytran732 commented:
Heyo :)

I am working on a lot of the migration too, so here are a few things I'd suggest :)

![image](https://user-images.githubusercontent.com/57488583/142737464-df854366-a29c-4228-8403-d88a5d30529d.png)

I think this part can be merged with the introduction at the top. I don't think the explanation for the 3 types of communications are necessary, we already touch on it below anyways. The top should just explain what end to end encryption is and how it removes the trust from the server - and by extension, the service provider, as they cannot read your messages anyways.

![image](https://user-images.githubusercontent.com/57488583/142737513-7b4a7160-ecef-4008-a0ee-c85176e64ca6.png)

I had a talk with dngray and we are moving away from using those alert banners, they do look pretty bad and it's not good that it is currently used just to tell people that they should move away from those chat apps without being a real alert. Besides, we already talks about the needs to remove trust from the service provider in the intro anyways, so it's redundant to have these.

![image](https://user-images.githubusercontent.com/57488583/142737571-b7374592-a947-45ce-afb9-c16f3f4d9916.png)

I think the btags are a bit inconsistent here.

![image](https://user-images.githubusercontent.com/57488583/142737590-52383aab-2e3d-49ca-9df4-6e3c04d3991a.png)

I think we should move away from the red alert tag as well. In my other PRs, I just explicitly put in the notes something in the line of: Hey, using Signal requires a phone number. Your chat messages are private, the metadata the server has is minimal, but you are not anonymous. Consider getting a disposable number depending on your threat model (linking the threat modeling page).

You can see an example of this on the https://privacyguides.org/video-streaming/ page (the LBRY section)

lrq3000 commented:
@tommytran732 Thank you so much for your very helpful feedback! Your other PR on browser is very helpful too and I used it as a guide for my own migration!

I currently have an issue since the merge, the layout is completely broken for some reason, so I'm trying to fix that first.

I will implement all your suggestions.

About the tags being inconsistent, what are you pointing at more specifically? The icons (this was done on purpose) or the missing background for "Audit" and "Whitepaper"? If the latter, this happens when there are both a link and a tooltip, then for some reason the current template removes the background. I would also prefer this to be fixed but I'm not Jekyll wizard enough to do that myself... Any help is welcome!

tommytran732 commented:
@lrq3000 Oh, I meant the missing background

And dw about the tests being stuck btw, dngray updated one of the dependencies and it broke github actions... he is trying to fix it xD

lrq3000 commented:
@tommytran732 thank you for the heads-up, but it was my mistake about the missing template ;-) It's now fixed fortunately :D

I have now implemented all your suggestions, except fixing the linked+tooltip labels missing background. Is the page now more in line with the planned new UI?

tommytran732 commented:
Yeah, well as you can see here:

![image](https://user-images.githubusercontent.com/57488583/142740581-dd93fc44-beb0-47ad-8222-51902945d292.png)


This area is a bit convoluted imo. I would shorten it, and actually merge it to the introduction, like the cloud storage page for example:

![image](https://user-images.githubusercontent.com/57488583/142740592-d77c8049-765d-4065-b4a2-7f63e2829b59.png)

Granted, I think we should mention that we only recommend open source tools, as that is one of the criteria. I think I should add that to the cloud storage page too tbh, once we are done wording it

tommytran732 commented:
![image](https://user-images.githubusercontent.com/57488583/142740636-5e1736ea-02ee-4efa-aea1-b9183ee89a88.png)

This is still inconsistent imo (lack of backgrouns)

I would also elaborate a bit more on Signal's note. Element looks good, Briar looks good. I think we should be mentioning the lack of forward secrecy in session, as that is one of the downsides of it.

I got 3 things I wanna talk about, but I feel like they should be separate discussions as they are not related to overhauling the UI, but:

- Status.im's last audit was in 2019, and there were several issues. IDK if those are rectified by now, I think we should do our homework on this, and if they aren't, should Status still be recommended?

- I feel like the recent news system is spammy, and personally I would remove them. But I think this needs discussions and approval

- Maybe we should look at https://molly.im?

Mikaela commented:
Do you plan to include https://github.com/privacyguides/privacyguides.org/issues/50 ?

dngray commented:
> @dngray I had a look at the new evergreen template thanks to the PR pointers you gave me. Wow, that's a great departure from the previous template! But it's great to see PG heading towards a more tutorial oriented approach, rather than listing. I understand the idea.

Basically that's the idea.

> I updated this PR to migrate to the evergreen template, please let me know what you think about it. I had to also migrate the labels subtemplate to allow the implementation of tags.

👍️

> I upgraded Status to a recommendation temporarily because I just moved everything to the evergreen template (and there is no Worth Mentions in it), I still need to test it out for myself.

It takes some getting used to.

> About removing the Videoconferencing section, I was referring to the fact that if independent audits are required for all recommended softwares, none of those in the videoconferencing section would pass, hence the section would be empty and hence would need to be removed. So what do we do about that?

We could put a mention that these things under Videoconferencing aren't yet auditing.

Re @tommytran732's post https://github.com/privacyguides/privacyguides.org/pull/192#issuecomment-974693966

I'm inclined to agree, as it allows us to keep the page clean. Also allows us to not have a zillion badges

I'm thinking for Audit we might make another button at the bottom, like we have for Privacy Policy.

> I currently have an issue since the merge, the layout is completely broken for some reason, so I'm trying to fix that first.

Yeah I'm going to rebase this off master, seems that one of the dependencies broke github actions.

lrq3000 commented:
@dngray Oh thank you so much for rebasing for me! I was going to do it but I just noticed that you took care of the issue. Ok yes I will rebase all my PRs from now on! I wasn't sure what you guys preferred. May I ask out of curiosity how did you rebase? Did you use `git rebase [-i] --no-ff P` as suggested [here](https://github.com/git/git/blob/master/Documentation/howto/revert-a-faulty-merge.txt) or was it another git wizardry? :-)

Thank you @dngray, @Mikaela , @tommytran732 for your feedbacks! I will implement them now and reply after :-)

Mikaela commented:
> Thank you dngray, @Mikaela , tommytran732 for your feedbacks! I will implement them now and reply after :-)

You're welcome. However [I left the team](https://mikaela.info/blog/english/2021/11/23/leaving-privacyguides.html) and am not planning on further reviews while I am happy to see #50 finally getting addressed.

lrq3000 commented:
Ok guys I have just implemented all your suggestions, although I had to make some different choices that I will explain below:
* @Mikaela about #50, I added all points as notes for Element instead of badges to limit the number of badges.
* @tommytran732 @dngray I decided to convert the Audit and Whitepaper badges into a new textual subsection "Audit" or "Audit & Whitepaper" where applicable, instead of a "Privacy Policy"-like button. I chose this to keep more expressivity, to detail what the audit is about: was the protocol audited, or the client software, or the server, or everything? This is I think a succinct enough way to display a crucial info for threat model assessment.

What remains in my todo is to assess Status more thoroughly. Please let me know if you have any other suggestion :-)

lrq3000 commented:
I forgot, @Mikaela , since the points you and other users mentioned aren't specific to Element but rather are broad issues for all decentralized platforms, I also added a short sentence to mention the possibility of missing of misworking features that are commonly expected in centralized apps, I hope this is OK for you.

tommytran732 commented:
Hey, The overall looks good, but I think we could elaborate a bit more on a few things :) 

Let's start with Signal first: 
In the description, maybe we can elaborate that the Signal protocol is widely praised and is also used in apps like WhatsApp.
Then, we should note that Signal stores your contact list encrypted using the user's PIN and that they do not have access to it.
Sealed sender should also be mentioned. We could elaborate a bit on how it reduces the metadata, and how it is only enabled by default for users who are already in the user's contact list. The user could enable sealed sender for everyone to reduce the metadata available to the Signal server, but then that would also subject them to potential spam ;)

lrq3000 commented:
@tommytran732 since none of the videoconferencing apps are audited, and it was the reason for your suggestion to remove Jami, since it seems we are going to include them nevertheless but with a mention that they are not audited, do you think we should add Jami back? It's not the best app but it's the only decentralized E2EE videoconferencing app (ie, E2EE group voice/video calls), the 2 others are centralized. Element is not E2EE when doing a group call.

tommytran732 commented:
> @tommytran732 since none of the videoconferencing apps are audited, and it was the reason for your suggestion to remove Jami, since it seems we are going to include them nevertheless but with a mention that they are not audited, do you think we should add Jami back? It's not a flawless app but it's the only decentralized E2EE videoconferencing app (ie, E2EE group voice/video calls), the 2 others are centralized. Element is not E2EE when doing a group call.

Do you use Jami? I mean, I guess... so long as it is actually usable and has no serious problems /shrug
I think we should probably get a team member to answer for the criteria tbh

lrq3000 commented:
@tommytran732 No I don't use Jami, I tried it once for the previous PR on PTIO but then dropped it. But note that I am not suggesting to add it back as an instant messenger, as all others are audited and hence much better choices, but as a videoconfering app this may be worthwhile... I need to research more how it encrypts video/voice calls.

About Signal, it's been a very long time since I used it, so I'm not up-to-date at all with its features. I will try to add your suggested changes but if you want more or a precise wording please feel free to suggest me what to add :-)

tommytran732 commented:
Ah yeah I would put those in the description instead of the notes, and I would ignore the MobileCoin part, since it doesn't affect privacy or security in any way

Mikaela commented on _data/software/messengers/2_element.yml:
```diff
@@ -0,0 +1,39 @@
+  <h4>Notes</h4>
+    <p>There are a few gotchas and missing features at the moment that need to be kept in mind to avoid unforeseen privacy breaches:</p>
+    <ul>
+        <li>Voice and video calls use the Jitsi library (not Jitsi Meet). Hence, 1-on-1 voice and video calls are end-to-end encrypted, but not group calls.</li>
```
What does this mean? Is it not using https://jitsi.riot.im ?

lrq3000 replied:
Not for 1-on-1 calls. I remember reading it was using lib-jitsi (the older jitsi implementation before jitsi meet) but maybe they moved to their own WebRTC implementation, I can't find the source I remember, it's buried somewhere in the tons of issues and docs they have. But I have a confirmation that only 1-on-1 calls are E2EE: https://github.com/vector-im/element-web/issues/12878

They expressed interest at implementing E2EE for group calls when Jitsi Meet's implementation will release as stable: https://github.com/vector-im/element-web/issues/13953

I'll come back later for the rest, and edit this text also to clarify.

Mikaela commented on _data/software/messengers/2_element.yml:
```diff
@@ -0,0 +1,39 @@
+    <p>There are a few gotchas and missing features at the moment that need to be kept in mind to avoid unforeseen privacy breaches:</p>
+    <ul>
+        <li>Voice and video calls use the Jitsi library (not Jitsi Meet). Hence, 1-on-1 voice and video calls are end-to-end encrypted, but not group calls.</li>
+        <li><a href="https://en.wikipedia.org/wiki/Forward_secrecy">Forward secrecy</a> can be enabled by the channel's creator (option: new users cannot access past history).</li>
```
I am not sure what do you mean with this. Do you mean room history visibility? That changes nothing on server database, which can still see it as it doesn't change prior messages. Or do you mean end-to-end encryption itself?

E2EE should be enabled by default when your client can see the other party having encryption capable devices/sessions, but that may not work every time either.

lrq3000 replied:
Yes, room history visibility. When the highest option is selected, this is an implementation of forward secrecy, so yes the messages are still stored on the database, but they are less likely to get bruteforcely decrypted by an attacker since each message is encrypted with a unique key.

Thank you for the reminder about E2EE by default, I just added a note.

Mikaela commented on _data/software/messengers/2_element.yml:
```diff
@@ -0,0 +1,39 @@
+        <li><a href="https://en.wikipedia.org/wiki/Forward_secrecy">Forward secrecy</a> can be enabled by the channel's creator (option: new users cannot access past history).</li>
+        <li>Deleting <a href="https://github.com/matrix-org/synapse/issues/1263">messages and media attachments</a> does not actually delete them from servers but just hide them along with the user stated reason, so that direct links still work. However, if room encryption was enabled, the messages cannot be decrypted by anyone but the intended users. Self-hosting does not resolve this <a href="h ...
+        <li>Reactions are not encrypted (<a href="https://github.com/matrix-org/matrix-doc/issues/2678">link to issue</a>).</li>
+        <li>Nicknames and avatars may leak even in private room (<a href="https://github.com/matrix-org/synapse/issues/5677">link to issue</a>).</li>
```
```suggestion
        <li>Nicknames and avatars may leak to users of other servers even in private rooms (<a href="https://github.com/matrix-org/synapse/issues/5677">link to issue</a>).</li>
```

Mikaela commented on _data/software/messengers/2_element.yml:
```diff
@@ -0,0 +1,39 @@
+    <ul>
+        <li>Voice and video calls use the Jitsi library (not Jitsi Meet). Hence, 1-on-1 voice and video calls are end-to-end encrypted, but not group calls.</li>
+        <li><a href="https://en.wikipedia.org/wiki/Forward_secrecy">Forward secrecy</a> can be enabled by the channel's creator (option: new users cannot access past history).</li>
+        <li>Deleting <a href="https://github.com/matrix-org/synapse/issues/1263">messages and media attachments</a> does not actually delete them from servers but just hide them along with the user stated reason, so that direct links still work. However, if room encryption was enabled, the messages cannot be decrypted by anyone but the intended users. Self-hosting does not resolve this <a href="h ...
```
I would just say non-text as text content will be removed 7 days from the sending by default settings.

Mikaela commented on _data/software/messengers/2_element.yml:
```diff
@@ -0,0 +1,39 @@
+    <ul>
+        <li>Voice and video calls use the Jitsi library (not Jitsi Meet). Hence, 1-on-1 voice and video calls are end-to-end encrypted, but not group calls.</li>
+        <li><a href="https://en.wikipedia.org/wiki/Forward_secrecy">Forward secrecy</a> can be enabled by the channel's creator (option: new users cannot access past history).</li>
+        <li>Deleting <a href="https://github.com/matrix-org/synapse/issues/1263">messages and media attachments</a> does not actually delete them from servers but just hide them along with the user stated reason, so that direct links still work. However, if room encryption was enabled, the messages cannot be decrypted by anyone but the intended users. Self-hosting does not resolve this <a href="h ...
+        <li>Reactions are not encrypted (<a href="https://github.com/matrix-org/matrix-doc/issues/2678">link to issue</a>).</li>
+        <li>Nicknames and avatars may leak even in private room (<a href="https://github.com/matrix-org/synapse/issues/5677">link to issue</a>).</li>
```
I wish these were actual warnings instead of just usual text.

lrq3000 replied:
Unfortunately it seems PG is parting ways with warnings and such, to adopt a more educational approach. As I understand it, the idea is to convey the risks/benefits in a more nuanced manner, which is good IMHO since no solution provide perfect privacy security.

The bright side I see is that now we don't have to argue about what to add or not in "risks", anything that is of interest to assess the threat model should be included IMHO. The disadvantage is indeed that we assume the reader is smart or curious enough to understand or research further the implications for their security.

dngray replied:
> Unfortunately it seems PG is parting ways with warnings and such, to adopt a more educational approach. As I understand it, the idea is to convey the risks/benefits in a more nuanced manner, which is good IMHO since no solution provide perfect privacy security.

Exactly, and a page with a bunch of warnings on everything leaves the user going. "So what do I use." uhh. or "what is best?"

Mikaela reviewed (changes requested):
Some comments on the Element part which is currently the only part of this PR I have interest towards.

lrq3000 commented:
I decided to rename the Whitepaper headings into Technical documentation, so that specifications published on github for example but not specifically termed "whitepapers" can be added. This applies for example to Briar. Because honestly there is no difference with a whitepaper apart from the name, both are self-published anyway.

lrq3000 commented:
Ok guys I have implemented everything that was mentioned so far.

I still have to test the two apps.

@tommytran732 I didn't add everything in the description of Signal because I feel they are not pertinent to most users, in my understanding the Notes subsections should contain any info worth mentioning to improve security or gotchas/pitfalls to know before using the app, but per se they aren't descriptive of the app's purpose. I however added that it's widespread and praised, and removed the criticisms about MobileCoin, which were just basic general criticism about cryptocurrencies by a well known critic (whose arguments are not holding much for now) and hence irrelevant about Signal's communication security despite what newspapers headlines tried to make out of it (and I fell for it...).

tommytran732 commented:
@lrq3000
1. I think you should remove the centralized tag from element. It is federated, not centralized.
2. Team Chat as a badge should be removed entirely - you can have group chat with Signal and Briar as well. It doesn't make any sense.
3. "Discover secure and private ways to communicate with others online without letting any third parties read your messages. " I think this is redundant and don't need to be mentioned.
4. "We only recommend communication softwares that support end-to-end encryption (E2EE), which ensures all communications are encrypted for anyone, including the server, except for you and the recipient. "
I think this could be phrased better, something along the lines of "The threat when using instant messenger is that the provider can look at your messages. Thus, we recommend only E2EE chats to mitigate this threat blah blah"
5. The deleted media section on element is too long. The described situation (federated servers can ignore the deletion request) applies to both the attached media and messages. You don't need to split messages to a different area.
6. Reactions, nicknames, avatar can be combined to 1 single section. Just mention that the only thing encrypted are the message body, nothing else. Metadata is fully available for anyone to see.
7. @dngray I think some links at the bottom could be cleaned up. Do you agree?

dngray commented:
Yes, sounds good, btw i'm going to rebase this PR soon on main.

ph00lt0 commented:
the links under "Complete Comparison" do not always have the right info and also they recommend apps that we certainly wouldn't, I would suggest not to include them on the privacyguides website.

dngray commented:
> @lrq3000
> 
> 1. I think you should remove the centralized tag from element. It is federated, not centralized.

Yes.

> 2. Team Chat as a badge should be removed entirely - you can have group chat with Signal and Briar as well. It doesn't make any sense.

Yes

> 3. "Discover secure and private ways to communicate with others online without letting any third parties read your messages. " I think this is redundant and don't need to be mentioned.

Correct, we don't recommend anything that doesn't have strong E2EE.

> 4. "We only recommend communication softwares that support end-to-end encryption (E2EE), which ensures all communications are encrypted for anyone, including the server, except for you and the recipient. "

> I think this could be phrased better, something along the lines of "The threat when using instant messenger is that the provider can look at your messages. Thus, we recommend only E2EE chats to mitigate this threat blah blah"

Yep, more explanatory and less technical. Says the same thing, maybe don't start the sentence with Thus, though. Just get straight into it.

I'd probably go more down the line of:

We only recommend messengers that support strong E2EE otherwise the chat platform administrators may be able to look at your messages. E2EE means your messages are encrypted before they are transmitted.

> 5. The deleted media section on element is too long. The described situation (federated servers can ignore the deletion request) applies to both the attached media and messages. You don't need to split messages to a different area.

Yup

> 6. Reactions, nicknames, avatar can be combined to 1 single section. Just mention that the only thing encrypted are the message body, nothing else. Metadata is fully available for anyone to see.

Yup

> 7. @dngray I think some links at the bottom could be cleaned up. Do you agree?

Agreed, I think we talked about it in the lounge, and it was determined many of them are quite out of date, might be worth moving it to the wiki rather than entirely deleting it. Might be useful for a blog article or something like that.

dngray commented:
> the links under "Complete Comparison" do not always have the right info and also they recommend apps that we certainly wouldn't, I would suggest not to include them on the privacyguides website.

Yes this, and i'd ditch the audits section. Each software recommendation should have that included in it's card.

dngray commented:
Rebased and squashed this to make it easier to work with.

dngray commented:
@tommytran732 and I decided not to mention Linphone/Jitsi as cards because they aren't audited. We do still mention Jitsi is used in the Element card.
